src/shogun/base/AnyParameter.h

```diff
@@ -151,6 +151,9 @@ namespace shogun
 		void set_value(const Any& value)
 		{
 			m_value = value;
+
+			for (auto& method : m_callback_functions)
+				method();
```

Comment:
wouldn't this be better in `CSGObject::update_parameter`? I think we either put all those things in SGObject or put all in AnyParameter, mixing it starts getting confusing.. Putting it here might actually be better in terms of performance? @vigsterkr @karlnapf

Reply:
do you think doing it in `Self::update` would be a good place?

Reply:
I am not sure actually, seems a bit heavy for something that just contains a map?

Reply:
`CSGObject::update_parameter` uses `Self::update`. If we to perform callback functions there, we would get the AnyParameter ourselves, and not use `Self::update`. Is this what you mean?

Reply:
yea, you'd need a getter to return a reference to the vector of callback functions, e.g. `self->map[_tag].get_callbacks()`. But I guess that makes it a bit ugly?

Reply:
nope, I think this type of separation of concerns is better